fe/fe-core/src/main/java/com/starrocks/connector/RemoteScanRangeLocations.java

```diff
@@ -319,25 +323,38 @@ private static int getSplitsIfBackendSplitFile(List<RemoteFileInfo> partitions,
     public List<TScanRangeLocations> getScanRangeLocations(DescriptorTable descTbl, Table table,
                                                            HDFSScanNodePredicates scanNodePredicates) {
         result.clear();
-        HiveMetaStoreTable hiveMetaStoreTable = (HiveMetaStoreTable) table;
 
         long start = System.currentTimeMillis();
+
+        class PartitionAttachment {
+            long partitionId = 0;
+            DataCacheOptions dataCacheOptions = null;
+        }
+
+        List<PartitionAttachment> partitionAttachments = new ArrayList<>();
         List<PartitionKey> partitionKeys = Lists.newArrayList();
         for (long partitionId : scanNodePredicates.getSelectedPartitionIds()) {
             PartitionKey partitionKey = scanNodePredicates.getIdToPartitionKey().get(partitionId);
             partitionKeys.add(partitionKey);
+            PartitionAttachment attachment = new PartitionAttachment();
+            attachment.partitionId = partitionId;
+            partitionAttachments.add(attachment);
         }
-        String catalogName = hiveMetaStoreTable.getCatalogName();
-        QualifiedName qualifiedName = QualifiedName.of(ImmutableList.of(catalogName,
-                hiveMetaStoreTable.getDbName(), hiveMetaStoreTable.getTableName()));
-        Optional<List<DataCacheOptions>> dataCacheOptionsList = generateDataCacheOptions(qualifiedName,
-                hiveMetaStoreTable.getPartitionColumnNames(), partitionKeys);
 
-        List<RemoteFileInfo> partitions;
+        Optional<List<DataCacheOptions>> dataCacheOptionsList = generateDataCacheOptions(table, partitionKeys);
+        if (dataCacheOptionsList.isPresent()) {
+            List<DataCacheOptions> dataCacheOptions = dataCacheOptionsList.get();
+            Preconditions.checkArgument(partitionAttachments.size() == dataCacheOptions.size());
+            for (int i = 0; i < partitionAttachments.size(); i++) {
+                partitionAttachments.get(i).dataCacheOptions = dataCacheOptions.get(i);
+            }
+        }
 
+        List<RemoteFileInfo> partitions;
         try {
-            GetRemoteFilesParams params = GetRemoteFilesParams.newBuilder().setPartitionKeys(partitionKeys).build();
-            partitions = GlobalStateMgr.getCurrentState().getMetadataMgr().getRemoteFiles(table, params);
+            GetRemoteFilesParams params = GetRemoteFilesParams.newBuilder().setPartitionKeys(partitionKeys)
+                    .setPartitionAttachments(partitionAttachments).build();
+            partitions = GlobalStateMgr.getCurrentState().getMetadataMgr().getRemoteFilesAsync(table, params).getAllOutputs();
```

Comment:
Why is it blocked here? Will it be made asyc here later?

Reply:
we will make it in next PRs.